File: pyforms_web/web/djangoapp/static/pyformsjs/ControlSlider.js
class ControlSlider extends ControlBase{

	
	////////////////////////////////////////////////////////////////////////////////

	get_value(){ 
		return this.jquery().val();
	};

	////////////////////////////////////////////////////////////////////////////////

	set_value(value){
		this.jquery().val(value);
		$( "#value"+this.name ).html( value );
	};

	////////////////////////////////////////////////////////////////////////////////

	init_control(){
		var html = 	"<div id='"+this.place_id()+"' class='field ControlSlider' title='"+this.properties.help+"' >";
		html += 	"<label>"+this.properties.label;
		html += 	" <div id='value"+this.control_id()+"' class='ui basic label'>"+this.properties.value+"</div>";
		html += 	"</label>";
		html += 	"<input style='width:100%;' type='range' name='"+this.name+"' value='"+this.properties.value+"' id='"+this.control_id()+"' min='"+this.properties.min+"' max='"+this.properties.max+"'>";
		html += 	"</div>";

		this.jquery_place().replaceWith(html);
		//this.jquery().on('input', function () {$(this).trigger('change');});
		var self = this;
		this.jquery().change(function(){ 
			$( "#value"+self.control_id() ).html( $(this).val() ); 
			self.basewidget.fire_event( self.name, 'changed_event' );
		});

		if(!this.properties.visible) this.hide();
	};

	////////////////////////////////////////////////////////////////////////////////

	deserialize(data){
		this.properties = $.extend(this.properties, data);
		this.set_value(this.properties.value);
	};

	////////////////////////////////////////////////////////////////////////////////

	serialize(){
		this.properties.value = this.get_value();
		return this.properties; 
	};

	////////////////////////////////////////////////////////////////////////////////
}

File: pyforms_web/controls/control_slider.py
from pyforms_web.controls.control_base import ControlBase
import simplejson

class ControlSlider(ControlBase):

    def __init__(self, *args, **kwargs):
        self._updateSlider = True
        self._min   = kwargs.get('min', 0)
        self._max   = kwargs.get('max', 100)
        if kwargs.get('default', None)==None:
            kwargs['default']=0
        super(ControlSlider,self).__init__(*args, **kwargs)
        
    def init_form(self): return "new ControlSlider('{0}', {1})".format( self._name, simplejson.dumps(self.serialize()) )


    @property
    def min(self): return self._min

    @min.setter
    def min(self, value):
        if self._min!=value:
            self.mark_to_update_client()
            self._min = value

    @property
    def max(self):  return  self._max

    @max.setter
    def max(self, value):
        if self._max!=value:
            self.mark_to_update_client()
            self._max = value

    def deserialize(self, properties):
        self.value    = int(properties.get('value',None))
        self._label   = properties.get('label','')
        self._help    = properties.get('help','')
        self._visible = properties.get('visible',True)

    def serialize(self):
        data = super(ControlSlider,self).serialize()
        data.update({ 'max': self.max, 'min': self.min, 'value': self._value })
        return data